vignettes/hys_law.Rmd

```diff
@@ -0,0 +1,156 @@
+
+the resulting dataset is helpful for deriving additional parameters.
+
+It may also prove useful for example, a listing where you have to display the two lab-records in one row to showcase the potential event. 
+
+```{r}
+altast_records <- adlb_annotated %>%
+  filter(PARAMCD %in% c("AST", "ALT"))
+
+bili_records <- adlb_annotated %>%
+  filter(PARAMCD %in% c("BILI"))
+
+hylaw_records <- derive_vars_joined(
+  dataset = altast_records,
+  dataset_add = bili_records,
+  by_vars = exprs(STUDYID, USUBJID, ADY),
+  order = exprs(ADY),
+  filter_join = ADT.join - ADT <= 14 & CRIT1FL == "Y" & CRIT1FL.join == "Y",
+  new_vars = exprs(BILI_LBSEQ = LBSEQ, BILI_DT = ADT, BILI_CRITFL = CRIT1FL),
+  mode = "first"
+)
+
+hylaw_records %>%
+  arrange(desc(BILI_CRITFL), desc(CRIT1FL)) %>%
+  dataset_vignette()
+```
+## How to Create New Paramters and Rows
+
+Using `derive_param_exist_flag()` you can create a variety of parameters for your final dataset with AVAL = 1/0 for your specific Hy's Law analysis. Below is an example of how to indicate a potential Hy's Law event, with `PARAMCD` set as `"HYSLAW"` and `PARAM` set to `"ALT/AST >= 3xULN and BILI >= 2xULN"` for **each patient** using the flags from the prior dataset. This method allows for flexibility as well, if parameters for each visit was desired, you would add `AVISIT` and `ADT` to the `select()` and `subject_keys` lines as denoted from the following code. 
+
+Additional modifications can be made such as:
+
+- Parameter to indicate worsening of condition
+- Any sort of baseline/post-baseline based analysis
+- Flags for other lab values like ALP if modified in above too
+
+```{r}
+hylaw_records_pts_visits <- hylaw_records %>%
+  select(STUDYID, USUBJID, TRT01A) %>% # add AVISIT, ADT for by visit
+  distinct()
+
+hylaw_records_fls <- hylaw_records %>%
+  select(STUDYID, USUBJID, TRT01A, CRIT1FL, BILI_CRITFL) %>% # add AVISIT, ADT for by visit
+  distinct()
+
+hylaw_params <- derive_param_exist_flag(
+  dataset_adsl = hylaw_records_pts_visits,
+  dataset_add = hylaw_records_fls,
+  condition = CRIT1FL == "Y" & BILI_CRITFL == "Y",
+  false_value = "N",
+  missing_value = "N",
+  subject_keys = exprs(STUDYID, USUBJID, TRT01A), # add AVISIT, ADT for by visit
+  set_values_to = exprs(
+    PARAMCD = "HYSLAW",
+    PARAM = "ALT/AST >= 3xULN and BILI >= 2xULN"
+  )
+)
+
+hylaw_params %>%
+  arrange(desc(AVAL)) %>%
+  dataset_vignette()
+```
+
+The last step would be binding these rows back to whatever previous dataset is appropriate based on your data specifications, in this case, it would be best suited to bind back to our `adlb_annotated` object.
+
+```{r}
+adlbhy <- adlb_annotated %>%
+  bind_rows(hylaw_params)
+```
+
+Hopefully this `adlbhy` object created is ready for your organization's standard macro display or code that is previously available for tables and listings related to this analysis. 
```

Comment:
I am very partial to conclusions.  High School Writing assignments drilled this into me.  Can we add a summary of what we did in the vignette and possibly how this is our first foray into this topic and welcome feedback and ideas